src/SignalR/common/Http.Connections/src/Internal/HttpConnectionsEventSource.cs

```diff
@@ -65,14 +73,42 @@ public ValueStopwatch ConnectionStart(string connectionId)
         [Event(eventId: 3, Level = EventLevel.Informational, Message = "Connection '{0}' timed out.")]
         public void ConnectionTimedOut(string connectionId)
         {
-            if (IsEnabled())
+            Interlocked.Increment(ref _connectionsTimedOut);
+
+            if (IsEnabled(EventLevel.Informational, EventKeywords.None))
             {
-                _connectionsTimedOut.WriteMetric(1.0f);
+                WriteEvent(3, connectionId);
+            }
+        }
 
-                if (IsEnabled(EventLevel.Informational, EventKeywords.None))
+        protected override void OnEventCommand(EventCommandEventArgs command)
+        {
+            if (command.Command == EventCommand.Enable)
+            {
+                // This is the convention for initializing counters in the RuntimeEventSource (lazily on the first enable command).
+                // They aren't disabled afterwards...
+
+                _connectionsStartedCounter ??= new PollingCounter("connections-started", this, () => _connectionsStarted)
                 {
-                    WriteEvent(3, connectionId);
-                }
+                    DisplayName = "Total Connections Started",
+                };
+                _connectionsStoppedCounter ??= new PollingCounter("connections-stopped", this, () => _connectionsStopped)
+                {
+                    DisplayName = "Total Connections Stopped",
+                };
+                _connectionsTimedOutCounter ??= new PollingCounter("connections-timed-out", this, () => _connectionsTimedOut)
+                {
+                    DisplayName = "Total Connections Timed Out",
+                };
+                _currentConnectionsCounter ??= new PollingCounter("current-connections", this, () => _currentConnections)
+                {
+                    DisplayName = "Current Connections",
+                };
+
+                _connectionDuration ??= new EventCounter("connections-duration", this)
+                {
+                    DisplayName = "Average Connection Duration",
```

Comment:
If we're renaming anyway, this should probably be called "connections-duration-milliseconds".

If we decide that's to verbose, we should still document the units somewhere. The DisplayName should include "milliseconds" or "ms" somewhere.

Reply:
There’s a new display units field coming